pennylane/circuit_graph.py

```diff
@@ -302,14 +255,25 @@ def ancestors(self, ops):
         Returns:
             list[Operator]: ancestors of the given operators
         """
-        # rx.ancestors() returns node indices instead of node-values
-        all_indices = set().union(*(rx.ancestors(self._graph, self._indices[id(o)]) for o in ops))
-        double_op_indices = set(self._indices[id(o)] for o in ops)
-        ancestor_indices = all_indices - double_op_indices
-
-        return list(self._graph.get_node_data(n) for n in ancestor_indices)
-
-    def descendants(self, ops):
+        if isinstance(ops, (Operator, MeasurementProcess)):
+            raise ValueError(
+                "CircuitGraph.ancestors accepts an iterable of"
+                " operators and measurements, not operators and measurements themselves."
+            )
+        if any(len(self._inds_for_objs[WrappedObj(op)]) > 1 for op in ops):
+            raise ValueError(
+                "Cannot calculate ancestors for an operator that occurs multiple times."
+            )
```

Comment:
If an operator occurs multiple times, wouldn't the ancestor be all ancestors of the last occurrence of this operator, minus all the previous occurrences?

Reply:
I think the issue is that its expected behaviour is ambiguous.